packages/calypso-e2e/src/lib/base-block.ts

```diff
@@ -0,0 +1,20 @@
+import { ElementHandle } from 'playwright';
+
+/**
+ * Base class for the Block type objects.
+ */
```

Comment:
This is definitely optional as it's just stylistic (and personal preference), but I wanted to throw it out there...

How do you feel about ditching the inheritance model here? With the new approaches to the `BaseContainer`, we're so close to being completely inheritance-free in this new Playwright infrastructure! 

My basic pitch is this: I think if we can avoid inheritance, it will help the code grow in a more readable and newcomer-friendly way. I think it can make these Block classes more explicit about how they work, and make it clearer for new folks on where they should add code (and avoiding split implementations, overrides, etc.).

The base constructor is just declaring what dependencies it needs (the element handle), and I think that's not too verbose to have each Block class just declare that, especially because the dependencies may vary across block (e.g., maybe a Block needs both a pointer to the Block element handle _and_ the Playwright `page`).

In the future, if we want shared functionality that applies to all blocks, I think we can make a separate set of library functions, and if needed have those library functions consume interfaces that we can then apply to relevant blocks. 

What do you think?

Reply:
Oh that's a good point, I will remove inheritance in the next commit.